src/engine/strat_engine/engine.rs

```diff
@@ -85,23 +87,14 @@ impl StratEngine {
         let mut table = Table::default();
         let mut incomplete_pools = HashMap::new();
 
-        for (pool_uuid, devices) in &pools {
-            match StratPool::setup(&dm, *pool_uuid, devices) {
+        for (pool_uuid, devices) in pools {
+            match setup_pool(&dm, pool_uuid, &devices, &table) {
                 Ok((pool_name, pool)) => {
-                    let evicted = table.insert(pool_name, *pool_uuid, pool);
-                    if evicted.is_some() {
-                        // TODO: update state machine on failure.
-                        let _ = teardown_pools(&dm, table);
-                        let err_msg = "found two pools with the same name";
-                        return Err(EngineError::Engine(ErrorEnum::Invalid, err_msg.into()));
-                    }
+                    table.insert(pool_name, pool_uuid, pool);
                 }
-                Err(e) => {
-                    // TODO: Add ability to distinguish expected vs. unexpected errors,
-                    // expected error would be caused by missing block device(s).
-                    warn!("Pool {} failed to setup, reason {:?}.", pool_uuid, e);
-                    // Unable to setup pool, it might become complete later.
-                    incomplete_pools.insert(*pool_uuid, devices.clone());
+                Err(err) => {
+                    warn!("pool failed to setup, reason {:?}", err);
```

Comment:
Is there some reason we need to remove the pool UUID from the message that is created?  Just knowing that some pool failed, isn't as helpful as knowing which pool failed and for what.

Reply:
Pool UUID is in the error returned by ```setup_pool()```.